src/workerd/api/web-socket.h

```diff
@@ -642,12 +645,25 @@ class WebSocket: public EventTarget {
   struct GatedMessage {
     kj::Maybe<kj::Promise<void>> outputLock;  // must wait for this before actually sending
     kj::WebSocket::Message message;
+    size_t pendingAutoResponses = 0;
   };
   using OutgoingMessagesMap = kj::Table<GatedMessage, kj::InsertionOrderIndex>;
   // Queue of messages to be sent. This is wraped in a IoOwn so that the pump loop can safely
   // access the map without locking the isolate.
   IoOwn<OutgoingMessagesMap> outgoingMessages;
 
+  // Keep track of current hibernatable websockets auto-response status to avoid racing
+  // between regular websocket messages, and auto-responses.
+  struct AutoResponse {
+    kj::Promise<void> ongoingAutoResponse = kj::READY_NOW;
+    kj::Table<kj::String, kj::InsertionOrderIndex> pendingAutoResponseQueue = kj::Table<kj::String, kj::InsertionOrderIndex>();
```

Comment:
Why a `kj::Table` for this and not, for instance, an `std::deque`? Given the `InsertionOrderIndex` I would prefer to use a more idiomatic structure here... but if this works, also non-blocking.

Reply:
I had a sense we avoided `std` whenever possible, but can make it a `std::deque` if that's preferred.

Reply:
Generally yes but if the shoe fits... There are a handful of places where we make use of `std::` containers, particularly queues, where they make sense.

Reply:
I once wrote a dub-linked list implementation in kj because I didn't know we could use `std::list` :sweat_smile: